tfhe/src/integer/gpu/server_key/radix/add.rs

```diff
@@ -323,13 +323,40 @@ impl CudaServerKey {
                 );
             }
         }
+        self.propagate_single_carry_assign_async(result, streams);
     }
 
-    pub fn unchecked_sum_ciphertexts(
+    pub fn unchecked_sum_ciphertexts<T: CudaIntegerRadixCiphertext>(
         &self,
-        ciphertexts: &[CudaUnsignedRadixCiphertext],
+        ciphertexts: &[T],
         streams: &CudaStreams,
-    ) -> Option<CudaUnsignedRadixCiphertext> {
+    ) -> T {
+        let mut result = self
+            .unchecked_partial_sum_ciphertexts_async(ciphertexts, streams)
+            .unwrap();
+        unsafe {
+            self.propagate_single_carry_assign_async(&mut result, streams);
+        }
+        streams.synchronize();
+        assert!(result.block_carries_are_empty());
+        result
+    }
+
+    pub fn unchecked_partial_sum_ciphertexts<T: CudaIntegerRadixCiphertext>(
+        &self,
+        ciphertexts: &[T],
+        streams: &CudaStreams,
+    ) -> Option<T> {
+        let result = self.unchecked_partial_sum_ciphertexts_async(ciphertexts, streams);
+        streams.synchronize();
+        result
+    }
+
+    pub fn unchecked_partial_sum_ciphertexts_async<T: CudaIntegerRadixCiphertext>(
```

Comment:
This function should be unsafe, as all async functions.

Reply:
done